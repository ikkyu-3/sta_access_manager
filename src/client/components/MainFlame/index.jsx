// @flow
import type { MainFlameProps } from "@types";

import * as React from "react";
import styles from "./style.css";

function MainFlame(props: MainFlameProps): React.Node {
  const { children } = props;
  const { main, mainFlame, standard } = styles;
  return (
    <main className={`${mainFlame} ${standard}`}>
      <div className={main}>{children}</div>
    </main>
  );
}
export default MainFlame;
