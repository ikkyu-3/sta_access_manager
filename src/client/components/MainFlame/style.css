@charset "utf-8";
.mainFlame {
  display: flex;
  position: relative;
  width: calc(100% - 170px);
  height: 100%;
  margin-left: 170px;
  box-sizing: border-box;
  padding: 40px;
}

.main {
  position: relative;
  width: 100%;
  height: 100%;
  background-color: white;
  color: gray;
  border: 4px solid lightgray;
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 0px 4px 1px grey;
  overflow: hidden;
}

.main > div {
  height: 100%;
}

/* デフォルト背景 */
.standard {
  background-color: #90ee90;
}

/* 仕事 */
.work {
  background-color: #bbfeff;
}

/* 自習 */
.study {
  background-color: #bcf3cb;
}

/* 勉強会 */
.meetUp {
  background-color: #f8f189;
}

/* サークル */
.circle {
  background-color: #f4d8f5;
}

/* 退出 */
.exit {
  background-color: #00558f;
}

/* エラー */
.error {
  background-color: #ea0032;
}
