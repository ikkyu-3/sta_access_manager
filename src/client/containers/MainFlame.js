// @flow
import { connect } from "react-redux";
import MainFlame from "@components/MainFlame";

export default connect(null)(MainFlame);
