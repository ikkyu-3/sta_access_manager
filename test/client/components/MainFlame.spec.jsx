import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import MainFlame from "@components/MainFlame";

describe("MainFlame.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(<MainFlame>children</MainFlame>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    it("子要素が表示されている", () => {
      expect(
        shallow(
          <MainFlame>
            <div>children</div>
          </MainFlame>
        )
          .find(".main")
          .contains(<div>children</div>)
      ).toBeTruthy();
    });
  });
});
