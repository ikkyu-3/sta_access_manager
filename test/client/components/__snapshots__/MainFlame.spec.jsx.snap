// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainFlame.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<main
  className="mainFlame standard"
>
  <div
    className="main"
  >
    children
  </div>
</main>
`;
